README.md
# -home-luigiamaducci-mysite-templates-login_page.html


history.txt
    1  cd mysite
    2  git status
    3  git add .
    4  git commit -m "l'informatique, quest-ce que vous faites? j'espere que le disque dur est bien branche"
    5  mkvirtualenv flask-tutorial --python=python3.6
    6  pip install flask flask-login flask-sqlalchemy mysql-connector-python
    7  pip freeze
    8  pip3.6 freeze
    9  pip freeze > requirements.txt
   10  git status
   11  git add .
   12  git commit -m "que son ame deborde de joie lorsqu'un pecheur se convertit"
   13  pip install Flask-Migrate
   14  export FLASK_APP=flask_app.py
   15  flask db init
   16  flask db migrate
   17  flask db migrate > thomas.txt
   18  flask db stamp head
   19  pip freeze > requirements.txt
   20  git status
   21  git add .
   22  git commit -m "la symbolique est souvent liee a celle de l'enfer"
   23  flask db migrate
   24  flask db upgrade
   25  git status
   26  git add .
   27  git commit -m "le pretre peut donne un lettre in articulo mortis"
   28  flask db migrate
   29  flask db upgrade
   30  flask shell
   31  echo "# -home-luigiamaducci-mysite-templates-login_page.html" >> README.md
   32  git init
   33  git add README.md
   34  git commit -m "first commit"
   35  git remote add origin https://github.com/jeanpix244/-home-luigiamaducci-mysite-templates-login_page.html.git
   36  git push -u origin master
   37  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="luigiamaducci",
    password="Exeter22",
    hostname="luigiamaducci.mysql.pythonanywhere-services.com",
    databasename="luigiamaducci$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "4565465456456465456456"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/1942c34abf5c_.py
"""empty message

Revision ID: 1942c34abf5c
Revises: a87c9265a732
Create Date: 2018-08-16 22:46:02.545531

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '1942c34abf5c'
down_revision = 'a87c9265a732'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/a87c9265a732_.py
"""empty message

Revision ID: a87c9265a732
Revises: 
Create Date: 2018-08-16 22:35:09.530674

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'a87c9265a732'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###
